lib/Jsrt/JsrtHelper.cpp

```diff
@@ -89,40 +91,32 @@ void JsrtCallbackState::ObjectBeforeCallectCallbackWrapper(JsObjectBeforeCollect
     }
 
 #ifndef _WIN32
-    static pthread_key_t s_threadLocalDummy;
+    static THREAD_LOCAL struct _OnThreadExit
+    {
+        ~_OnThreadExit()
+        {
+            ThreadBoundThreadContextManager::DestroyContextAndEntryForCurrentThread();
+        }
+    } _s_onThreadExit;
 #endif
-    static THREAD_LOCAL bool s_threadWasEntered = false;
 
-    _NOINLINE void DISPOSE_CHAKRA_CORE_THREAD(void *_)
```

Comment:
BTW; prev. pthread approach was guaranteed to run at the `last phase` of thread exit. Current approach doesn't have a priority in place. That makes me wonder that; If altering the timing for `ThreadBoundThreadContextManager::DestroyContextAndEntryForCurrentThread()` call has improved the test results positively, we might have a `race` issue on cleanup that we might have to target.